core/src/main/java/com/linecorp/armeria/client/ClientFactoryOptions.java

```diff
@@ -0,0 +1,367 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import static java.util.Objects.requireNonNull;
+
+import java.net.InetSocketAddress;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterables;
+
+import com.linecorp.armeria.common.CommonPools;
+import com.linecorp.armeria.common.Flags;
+import com.linecorp.armeria.common.util.AbstractOptions;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Metrics;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.EventLoop;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.epoll.EpollChannelOption;
+import io.netty.handler.ssl.SslContextBuilder;
+import io.netty.resolver.AddressResolverGroup;
+import it.unimi.dsi.fastutil.objects.Object2ObjectArrayMap;
+
+/**
+ * A set of {@link ClientFactoryOption}s and their respective values.
+ */
+@SuppressWarnings("OptionalGetWithoutIsPresent")
+public final class ClientFactoryOptions extends AbstractOptions {
+    private static final EventLoopGroup DEFAULT_WORKER_GROUP = CommonPools.workerGroup();
+
+    private static final Function<? super EventLoopGroup, ? extends EventLoopScheduler>
+            DEFAULT_EVENT_LOOP_SCHEDULER_FACTORY =
+            eventLoopGroup -> new DefaultEventLoopScheduler(eventLoopGroup, 0, 0, Collections.emptyList());
```

Comment:
I think this should be `eventLoopGroup, 1, 0, Collections.emptyList()` so that only one connection is made for HTTP/2

Reply:
I passed `0` to (implicitly) express that it will follow the default value. (The previous `ClientFactoryBuilder` also set `0` if `maxNumEventLoopsPerEndpoint` is not specified.) It will eventually use `1` (`DEFAULT_MAX_NUM_EVENT_LOOPS`) inside `DefaultEventLoopScheduler`.

I understand `1` for `maxNumEventLoopsPerEndpoint` is more semantically right. IMO, `DEFAULT_MAX_NUM_EVENT_LOOPS` should be used instead of `1`, then.

Reply:
Ah, yes. I completely forgot about it. 😱

Reply:
Let me send the fixing PR. Thanks!